moveit_simple/src/moveit_simple.cpp

```diff
@@ -611,54 +619,233 @@ void Robot::clearTrajectory(const::std::string traj_name)
         throw IKFailException("Conversion to joint trajectory failed for " + traj_name);
       }
     }
     catch(CollisionDetected &cd)
     {
-      ROS_ERROR_STREAM("Collision detected in the " << traj_name);
+      ROS_ERROR_STREAM("Collision detected in [" << traj_name << "]");
       throw cd;
     }
   }
   else
   {
-    ROS_ERROR_STREAM("Trajectoy " << traj_name << " not found");
+    ROS_ERROR_STREAM("Trajectory [" << traj_name << "] not found");
     throw std::invalid_argument("No trajectory found named " + traj_name);
   }
 }
 
 
 
+void OnlineRobot::execute(const std::string traj_name, bool collision_check)
+{
+  std::lock_guard<std::recursive_mutex> guard(m_);
+
+  try 
+  {
+  	std::vector<moveit_simple::JointTrajectoryPoint> goal;
+
+  	goal = plan(traj_name, collision_check);
+  	execute(goal);
+  }
+  catch	(IKFailException &ik)
+  {
+  	ROS_ERROR_STREAM("IK Failed for trajectory: [" << traj_name << "]");
+  	throw ik;
+  }
+  catch	(CollisionDetected &cd)
+  {
+  	ROS_ERROR_STREAM("Collision detected in trajectory");
+	throw cd;
+  }
+  catch (std::invalid_argument &ia)
+  {
+    ROS_ERROR_STREAM("Invalid trajectory name: [" << traj_name << "]");
+    throw ia;
+  }
+  catch (ExecutionFailureException &ef)
+  {
+    ROS_ERROR_STREAM("Trajectory [" << traj_name << "] failed to execute");
+    throw ef;
+  } 
+}
+
+
+
+void OnlineRobot::execute(std::vector<moveit_simple::JointTrajectoryPoint> & joint_trajectory_points,
+                          bool collision_check)
+{
+  std::lock_guard<std::recursive_mutex> guard(m_);
+
+  const double TIMEOUT_SCALE = 1.25;  //scales time to wait for action timeout.
+  
+  int collision_points = 0;
+
+  control_msgs::FollowJointTrajectoryGoal goal = 
+  				trajectoryTypeConversionFromMoveItSimpleToROS(joint_trajectory_points);
+
+  if (!jointTrajectoryCollisionCheck(goal, collision_points, collision_check) )
+  {
+
+    ros::Duration traj_time =
+        goal.trajectory.points[goal.trajectory.points.size()-1].time_from_start;
+    ros::Duration timeout(TIMEOUT_SCALE * traj_time.toSec());
+    if (action_.sendGoalAndWait(goal, timeout) == actionlib::SimpleClientGoalState::SUCCEEDED)
+    {
+      ROS_INFO_STREAM("Successfully executed Joint Trajectory ");
+    }
+    else
+    {
+      ROS_ERROR_STREAM("Joint Trajectory failed to execute.. check input Plan");
+      throw ExecutionFailureException("Execution failed for Joint Trajectory");
+    }
+  }
+
+  else
+  {
+  	ROS_ERROR_STREAM("Collision detected at " << collision_points << 
+                       " points for Joint Trajectory");
+
+  	throw CollisionDetected("Collision detected while interpolating ");
+  }
+}
+
+
+
+std::vector<moveit_simple::JointTrajectoryPoint> Robot::trajectoryTypeConversionFromROSToMoveItSimple(std::vector<trajectory_msgs::JointTrajectoryPoint> & ROS_joint_trajectory_points)  const
+{
+	std::vector<moveit_simple::JointTrajectoryPoint> goal;
```

Comment:
Don't change, but since you know the size of the trajectory you are converting, you may consider calling `reserve` ahead of time for better performance.